### PR TITLE
Add support for argument separator '--'

### PR DESCRIPTION
**Pull request by natemcmaster**
Add support for using '--' to separate command args from remaining args. Use case: dotnet-watch or dotnet-run.

This will put everything following the first '--' into `app.RemainingArguments`

cc @pranavkm @victorhurdugaci

### CODE REVIEW

File: src/Microsoft.Extensions.CommandLineUtils/CommandLine/CommandLineApplication.cs [hunk 1]
```diff
@@ -146,10 +147,17 @@ public int Execute(params string[] args)
                     if (longOption != null)
                     {
                         processed = true;
-                        option = command.GetOptions().SingleOrDefault(opt => string.Equals(opt.LongName, longOption[0], StringComparison.Ordinal));
+                        var longOptionName = longOption[0];
+                        option = command.GetOptions().SingleOrDefault(opt => string.Equals(opt.LongName, longOptionName, StringComparison.Ordinal));
 
                         if (option == null)
                         {
+                            if (string.IsNullOrEmpty(longOptionName) && !command._throwOnUnexpectedArg  && AllowArgumentSeparator)
```
Comment (victorhurdugaci) [inline src/Microsoft.Extensions.CommandLineUtils/CommandLine/CommandLineApplication.cs]: Does it allow any number of `--`

Comment (natemcmaster) [inline src/Microsoft.Extensions.CommandLineUtils/CommandLine/CommandLineApplication.cs]: It puts everything after the first '--' in app.RemainingArguments.

Example: if specified twice, `dotnet command -- --`, then `app.RemainingArguments == new [] { "--" }`

Comment (victorhurdugaci) [inline src/Microsoft.Extensions.CommandLineUtils/CommandLine/CommandLineApplication.cs]: Nice!

**Comment by natemcmaster** [comment]
cc @victorhurdugaci regardless of whether https://github.com/aspnet/dotnet-watch/pull/159 goes in or not, does this PR look okay?

**Comment by victorhurdugaci** [comment]
Yes, it's good :shipit:

**Comment by natemcmaster** [comment]
😎 cool

**Pull request merged**